src/app.jsx

```diff
@@ -1,19 +1,21 @@
-import React, { useState } from 'react'
-import useLocalStorage from 'react-use/lib/useLocalStorage'
+import React, { useState, useContext } from 'react'
 import AddDiceModal from './components/addDiceModal'
 import Dice from './components/dice'
 import { dispatch } from './hooks/useBus'
+import DicesContext from './contexts/dices'
 import './app.css'
 
 const App = () => {
   const [printModal, setPrintModal] = useState(false)
 
-  const [dices, setDices] = useLocalStorage('dices', [])
-  const reset = () => setDices([])
-  const roll = () => dispatch('@@ui/ROLL>DICES')
+  const { dices, reset, add } = useContext(DicesContext)
 
   const addDice = (faces) => {
-    setDices([...dices, { id: Date.now(), faces }])
+    add({
```

Comment:
I don't understand the purpose of the handler `addDice`.
why the hook cannot handle the `id` setting ?

Reply:
1. `addDice` helps me 
  a. add a dice to the context
  b. close the modal

2. I can move the id into `dicesContext.add`, in which case it should returns the added dice

Reply:
I missed the close modal part ^^
I prefer this way where the `dicesContext.add` takes care of everything related to dice modifications